### DISCUSSION TITLE
why the line layer does not shown as invisible or deck?

### DISCUSSION THREAD

**Discussion by qiaoxin136** [visgl/react-google-maps#240] [Q&A]
I follow the example using GoogleMapsOverlay, i set up a checkbox to turn on and off the line layer. please the codesandbox, 

https://codesandbox.io/p/github/qiaoxin136/testgooglemaps/main?file=%2Fsrc%2FApp.tsx&workspaceId=d68df01c-f46f-497c-bfee-4134267b36a9

i console out the deck, when i change the checkbox from "off" to "on" to "off", the consoled out deck is shown as "false", "true", "false", but the line layer, shown as "invisible", "visible", "visible". I need your help to understand why i can not turn on/off the line layer as i desired. 

see the pic below, 

![image](https://github.com/visgl/react-google-maps/assets/44871863/af0ef0da-45d2-43e3-a883-f31fda7101fd)

**Comment by usefulthink** [comment]
I only get a "Unable to access this workspace" message when I try to visit the codesandbox, and without that it's impossible to tell what is happening there.

**Comment by qiaoxin136** [comment]
sorry, codesandbox changed. I do not know how to fix that. I put it into stackblitz. here is the link. please try this. if it does not work, please let me know. Really appreciate your help. 


https://stackblitz.com/~/github.com/qiaoxin136/testgooglemaps

**Comment by usefulthink** [comment]
There was a missing effect cleanup function, so we were likely looking at an older version of the overlay that was never removed...

```tsx
const DeckGlOverlay = ({ layers }: DeckglOverlayProps) => {
  const deck = useMemo(() => new GoogleMapsOverlay({ interleaved: true }), []);
  const map = useMap();

  useEffect(() => {
    deck.setMap(map);

    // this was missing:
    return () => deck.setMap(null);
  }, [deck, map]);

  useEffect(() => {
    deck.setProps({ layers });
  }, [deck, layers]);

  return null;
};
```

**Reply by usefulthink** [reply]
I'll make sure to update our documentation to add that as well.

**Reply by qiaoxin136** [reply]
i got it. thank you for clarifying.
